src/_P118_Itho.ino

```diff
@@ -0,0 +1,548 @@
+			Device[deviceCount].InverseLogicOption = false;
+			Device[deviceCount].FormulaOption = false;
+			Device[deviceCount].ValueCount = 3;
+			Device[deviceCount].SendDataOption = true;
+			Device[deviceCount].TimerOption = true;
+			Device[deviceCount].GlobalSyncOption = true;
+		  break;
+		}
+
+	case PLUGIN_GET_DEVICENAME:
+		{
+			string = F(PLUGIN_NAME_118);
+			break;
+		}
+
+	case PLUGIN_GET_DEVICEVALUENAMES:
+		{
+			strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[0], PSTR(PLUGIN_VALUENAME1_118));
+			strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[1], PSTR(PLUGIN_VALUENAME2_118));
+			strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[2], PSTR(PLUGIN_VALUENAME3_118));
+			break;
+		}
+
+	case PLUGIN_GET_DEVICEGPIONAMES:
+	  {
+			event->String1 = formatGpioName_input(F("Interrupt pin (CC1101 GDO2)"));
+			break;
+    }
+
+	case PLUGIN_SET_DEFAULTS:
+		{
+    	PCONFIG(0) = 1;
+			PCONFIG(1) = 10;
+			PCONFIG(2) = 87;
+			PCONFIG(3) = 81;
+    	success = true;
+			break;
+		}
+
+	case PLUGIN_INIT:
+		{
+			//If configured interrupt pin differs from configured, release old pin first
+			if ((Settings.TaskDevicePin1[event->TaskIndex]!=Plugin_118_IRQ_pin) && (Plugin_118_IRQ_pin!=-1))
+			{
+				addLog(LOG_LEVEL_DEBUG, F("IO-PIN changed, deatachinterrupt old pin"));
+				detachInterrupt(Plugin_118_IRQ_pin);
+			}
+			LoadCustomTaskSettings(event->TaskIndex, (byte*)&PLUGIN_118_ExtraSettings, sizeof(PLUGIN_118_ExtraSettings));
+			addLog(LOG_LEVEL_INFO, F("Extra Settings PLUGIN_118 loaded"));
+			PLUGIN_118_rf.setDeviceID(PCONFIG(1), PCONFIG(2), PCONFIG(3)); //DeviceID used to send commands, can also be changed on the fly for multi itho control, 10,87,81 corresponds with old library
+			PLUGIN_118_rf.init();
+			Plugin_118_IRQ_pin = Settings.TaskDevicePin1[event->TaskIndex];
+			pinMode(Plugin_118_IRQ_pin, INPUT);
+			attachInterrupt(Plugin_118_IRQ_pin, PLUGIN_118_ITHOinterrupt, FALLING);
+			addLog(LOG_LEVEL_INFO, F("CC1101 868Mhz transmitter initialized"));
+			PLUGIN_118_rf.initReceive();
+			PLUGIN_118_InitRunned=true;
+			success = true;
+			break;
+		}
+
+	case PLUGIN_EXIT:
+	{
+		addLog(LOG_LEVEL_INFO, F("EXIT PLUGIN_118"));
+		//remove interupt when plugin is removed
+		detachInterrupt(Plugin_118_IRQ_pin);
+		success = true;
+		break;
+	}
+
+  case PLUGIN_ONCE_A_SECOND:
+  {
+		//decrement timer when timermode is running
+		if (PLUGIN_118_State>=10) PLUGIN_118_Timer--;
+
+		//if timer has elapsed set Fan state to low
+		if ((PLUGIN_118_State>=10) && (PLUGIN_118_Timer<=0))
+		{
+			PLUGIN_118_State=1;
+			PLUGIN_118_Timer=0;
+		}
+
+		//Publish new data when vars are changed or init has runned or timer is running (update every 2 sec)
+		if  ((PLUGIN_118_OldState!=PLUGIN_118_State) || ((PLUGIN_118_Timer>0) && (PLUGIN_118_Timer % 2==0)) || (PLUGIN_118_OldLastIDindex!=PLUGIN_118_LastIDindex)|| PLUGIN_118_InitRunned)
+		{
+			addLog(LOG_LEVEL_DEBUG, F("UPDATE by PLUGIN_ONCE_A_SECOND"));
+			PLUGIN_118_Publishdata(event);
+			sendData(event);
```

Comment:
If you use your own timer/interval mechanism, shouldn't the `Device[n].TimerOptional = true;` flag be added to the device flags? Now it will be triggered in 2 separate intervals.

Reply:
Added, I also disabled the timer option as it has no real use.